packages/core/src/core.ts

```diff
@@ -635,31 +654,37 @@ class ThresholdKey implements ITKey {
     importedKey?: BN;
     delete1OutOf1?: boolean;
   } = {}): Promise<InitializeNewKeyResult> {
-    if (!importedKey) {
-      const tmpPriv = generatePrivate();
+    let seed: Uint8Array = new Uint8Array();
+    if (this.keyType === KeyType.secp256k1) {
+      const tmpPriv = importedKey || generatePrivate(this.keyType);
       this._setKey(new BN(tmpPriv));
+    } else if (this.keyType === KeyType.ed25519) {
+      seed = importedKey ? new Uint8Array(importedKey.toBuffer(undefined, 32)) : nacl.randomBytes(32);
+      const keyPair = nacl.sign.keyPair.fromSeed(seed);
+      const tempPriv = new BN(keyPair.secretKey.slice(0, 32)).umod(this.ecCurve.curve.n);
+      this._setKey(tempPriv);
```

Comment:
This looks odd.

`keyPair.secretKey.slice(0, 32) == seed`

Are we sure we want the `privKey` to be the seed? Shouldn't it be the derived scalar instead?

@metalurgical @himanshuchawla009

Reply:
Or it doesn't matter? But then why not just write `new BN(seed).umod(this.ecCurve.curve.n)`?

Also, should we make the `importedKey` of type `Buffer` or `Uint8Array` and require it to be 32 bytes?

Reply:
i have fixed all this in my PR, plz wait for 30 mins before reviewing, facing some typescript issues, which is blocking me to merge the PR